libraries/AP_Baro/AP_Baro_SPL06.cpp

```diff
@@ -135,12 +165,25 @@ bool AP_Baro_SPL06::_init()
     _c20 = ((uint16_t)buf[12] << 8) | (uint16_t)buf[13];
     _c21 = ((uint16_t)buf[14] << 8) | (uint16_t)buf[15];
     _c30 = ((uint16_t)buf[16] << 8) | (uint16_t)buf[17];
+    if(type == Type::SPA06) {
+		_c31 = (buf[18] & 0x80 ? 0xF000 : 0) | ((uint16_t)buf[18] << 4) | (((uint16_t)buf[19] & 0xF0) >> 4);
+		_c40 = ((buf[19] & 0x8 ? 0xF000 : 0) | ((uint16_t)buf[19] & 0x0F) << 8) | (uint16_t)buf[20];
+	}
 
     // setup temperature and pressure measurements
     _dev->setup_checked_registers(3, 20);
 
+#ifdef SPL06_BACKGROUND_ENABLE
```

Comment:
```suggestion
#if AP_BARO_SPL06_BACKGROUND_ENABLE
```

And ensure this is always define